Lib/functools.py

```diff
@@ -882,6 +882,7 @@ def __init__(self, func):
         self.func = func
         self.attrname = None
         self.__doc__ = func.__doc__
+        self.__isabstractmethod__ = getattr(func, '__isabstractmethod__', False)
```

Comment:
Would it be possible to only set the attribute (on self) if the attribute exists on func?